src/node-lib/curriculum-api-2023/shared.schema.new.ts

```diff
@@ -0,0 +1,263 @@
+import { z } from "zod";
+
+/**
+ *
+ * - This should directly map to its equivalent in OWA
+ * - The intention is for this file to be moved to an npm package that can be shared between DBT and OWA
+ *
+ */
+
+export const _stateSchema = z.enum(["published", "new", "migration"]);
+export const _cohortSchema = z.enum(["2020-2023", "2023-2024", "0"]);
+
+export const lessonDataSchema = z.object({
+  lesson_id: z.number(),
+  lesson_uid: z.string(),
+  slug: z.string(),
+  title: z.string(),
+  description: z.string(),
+  pupil_lesson_outcome: z.string(),
+  key_learning_points: z.array(z.object({})),
```

Comment:
do these types parse ok as empty objects?

Reply:
yes because zod doesn't treat them as empty objects . it just doesn't look at any fields inside the object. It could definitely be better defined though.